### ISSUE TITLE
Empty attachment sent if message body is empty

### ISSUE THREAD

**Issue by yujiri8** [domodwyer/mailyak#61]
When I send an email with mailyak, the inbox entry in GMail shows it as having a 0-byte attachment called 'noname'. It doesn't happen with other email clients. Using the "show original" link in GMail, I found that the raw message sent by mailyak is different from the ones sent by the builtin net/smtp or by mutt; after the basic headers, there is an extra:
```
Content-Type: multipart/mixed; boundary="2a0c83baa5930f530706af609167363873a970accfe2546bbe7b21fc5aa5"; charset=UTF-8

--2a0c83baa5930f530706af609167363873a970accfe2546bbe7b21fc5aa5
Content-Type: multipart/alternative; boundary="6033160c56daa2f78449cfadf52e90cefb89f10bdf665734d5744c2342ac"

--6033160c56daa2f78449cfadf52e90cefb89f10bdf665734d5744c2342ac--
--2a0c83baa5930f530706af609167363873a970accfe2546bbe7b21fc5aa5--
```
Which must be what's causing GMail to interpret it as an empty, nameless attachment.

**Comment by domodwyer** [comment]
Hi @yujiri8 

That sounds like a bug! Do you have the example code to produce the above email? Just wondering if there's anything unusual to try narrow it down.

I will look into this at the weekend, I can't imagine it's going to be too difficult - **THANK YOU** for reporting it!

Dom

**Comment by yujiri8** [comment]
Taken straight from the README, reduced a bit:
```go
package main

import (
	"net/smtp"

	"github.com/domodwyer/mailyak"
)

func main() {
	// Create a new email - specify the SMTP host:port and auth (if needed)
	mail := mailyak.New("mail.host.com:25", smtp.PlainAuth("", "user", "pass", "mail.host.com"))

	mail.To("dom@itsallbroken.com")
	mail.From("oops@itsallbroken.com")

	// And you're done!
	if err := mail.Send(); err != nil {
		panic(" 💣 ")
	}
}
```

**Comment by domodwyer** [comment]
Hi @yujiri8!

Thanks for reporting this - I've pushed a fix and cut a release `v3.3.2` - I hope this fixes any problems you were having :)

**Thanks** for reporting! It's much appreciated

Dom
